app/Main.hs

```diff
@@ -44,9 +44,10 @@ getPlayer playerNumber gameState = case playerNumber of
 
 
 chooseFlag :: PlayerNumber -> GameState -> IO Flag
-chooseFlag = undefined
-
+chooseFlag playerNumber gameState = toEnum <$> (read <$> ask "please pick a flag")
```

Comment:
Additionally, I'm sure you're aware this isn't type safe since `read` is a partial function. You might consider trying to solve this without using partial functions.